server/routes/index.ts

```diff
@@ -39,23 +42,92 @@ router.post("/login", async (req, res) => {
   try {
     const user = await User.findOne({ username });
 
-    if (!user || user.password !== password) {
+    if (!user) {
+      return res.status(401).json({ message: "Invalid username or password" });
+    }
+
+    if (password === undefined) {
+      return res.status(401).json({ message: "Password is undefined" });
+    }
+
+    const isPasswordValid = await Password.verify(password, user.password);
+    if (!isPasswordValid) {
       return res.status(401).json({ message: "Invalid username or password" });
     }
+    const secretKey = process.env.JWT_SECRET;
+
+    // Tiếp tục xử lý đăng nhập thành công
+    const token = jwt.sign({ userId: user._id }, secretKey, {
+      expiresIn: "5m",
+    });
+
+    res.json({ message: "Login successful", user, token });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "An error occurred" });
+  }
+});
+// Tạo và lưu trữ Refresh Token khi người dùng đăng nhập
+const createRefreshToken = async (userId) => {
```

Comment:
Hàm này nên là private, refreshToken thì chỉ cần dùng hàm random để sinh ra 1 chuỗi ký tự unique là đủ 
https://stackoverflow.com/questions/43837659/guid-uuid-in-typescript-node-js-app